app/assets/javascripts/controllers/dialog_editor/dialog_editor_controller.js

```diff
@@ -105,10 +106,24 @@ ManageIQ.angular.app.controller('dialogEditorController', ['$window', '$http', '
     if (vm.treeSelectorIncludeDomain === false) {
       fqname.splice(1, 1);
     }
-    elementData.resource_action.ae_namespace = fqname.join('/');
+    elementData.resource_action.ae_instance = fqname.pop();
+    elementData.resource_action.ae_class = fqname.pop();
+    elementData.resource_action.ae_namespace = fqname.filter(String).join('/');
     vm.treeSelectorShow = false;
   }
 
+  function showFullyQualifiedName(resourceAction) {
+    if (typeof resourceAction.ae_namespace == 'undefined' ||
+        typeof resourceAction.ae_class == 'undefined' ||
+        typeof resourceAction.ae_instance == 'undefined') {
+      return '';
+    }
+    var fqname = resourceAction.ae_namespace
```

Comment:
It's better to use lodash's reduce function here (if resourceAction contains only items which can be merged together)
```javascript
return _.reduce(resourceAction,
  function (curr, item) {
    return item + '/' + curr;
  }, '');
```

Reply:
@karelhala `resourceAction` contains more items than just these that I'm merging together in `showFullyQualifiedName` method.